### PR TITLE
Feature/variousCrudeViews

### LINKED ISSUE #2: Testing Header, Login, Manager, LandingPage, App
* **I'm submitting a ...**
  - [ ] bug report
  - [ ] feature request
  - [X] support request 

* **Do you want to request a *feature* or report a *bug*?**
Fetch functionality not executed, and do not expect test methods involving fetch to be mocked right now. 

* **What is the current behavior?**

Test currently rendering components: 
`Header`, `Login`, `Manager`, `LandingPage`, `App` components. 

Expect the tests to be modified down the line if components change.

* **What is the expected behavior?**
All tests should pass. Tests should look for rendered data and fired events.

* **What is the motivation / use case for changing the behavior?**
TDD, baby! 
In all seriousness, testing these methods helped me catch bugs, and motivates me to dive deeper into testing for ultimate confidence in my code.

### PR DESCRIPTION
**Pull request by leighlars**
##OverLook Pull Request

### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Testing - building tests
- [x] Formatting - no new features
- [X] Styling- no new features

### Detailed Description 
- Why is this change required? 

This PR initiates some early steps to get us going. 
Fetch is not implemented, but I have peppered in some commented-out methods from the last project throughout the components as a fallback. Feel free to delete if you have a better solution.

We can render the various pages, and their appropriate child components, except for Guest. See issue below. 
I've added a fireEvent test for Login NavLink test in `App`, Login form input test in `Login` and MgrSearch form input test in `MgrSearchForm`. The tests for mocked methods are skipped because I didn't write the functionality for any methods. Again, just a bunch of shells of functions everywhere. 🐢

isLoggedIn is no longer a state. Instead, certain components will render in `View` based on the user. Thought it was cleaner and made more sense. 

Added conditional to render a welcome greeting to user in top right corner.

Added a MgrSearchForm child to Manager component, not a fan of the name. Whoever decides to change it will need to make sure every other file and imported name into `Manager` will need to be changed, too. 

Housekeeping: added a label for 'UX' in GH projects. 

- What problem does it solve? 
It gets us going. Whoever does the API stuff at this point is gonna get this thing really off the ground. And of course the Guest view issue. 

- Any challenges that arose while implementing this feature? 

I am unable to render a guest page in `View`. I've removed 'manager' and 'null' from `this.state.user` value, and used 'Sarah" as the `user.name` value but it's throwing me an error that user is not defined. 

- If so, how were the addressed?

I changed it back so that the error will go away, but I look forward to someone finding a solution for this error. 

### Components modified:
- [X] App
- [X] Login
- [ ] Guest
- [X] Header
- [X] LandingPage
- [X] Manager
- [ ] other files:  the test and scss files for these components

### Any Additional Information for the team
How can we cleanly assign each user their appropriate bookings after Fetch? Would this happen in `CleanData`, or in `App`?
Not sure how to test for `Error`, skipped that test in last project. 
I added in as many notes as I could think of in regards to testing and functionality throughout the codebase. 
Check out commits to see the "story", or look at the files to see the code changes.

### CODE REVIEW

File: src/App/App.test.js [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
+   const { getByRole } = render(<MemoryRouter> <App /></MemoryRouter>)
+   const loginLink = screen.getByRole("link", { name: "Login" })
+  //  check login link in header
+   fireEvent.click(loginLink);
+   // go to log in login page
+   // render login form
+   expect(screen.getByPlaceholderText("Username")).toBeInTheDocument()
+   const loginFormBtn = screen.getByRole("button", { name: "Login" })
+  });
+
```
Comment (jordy1611) [inline src/App/App.test.js:37]: Can definitely get more specific with the queries. This is a good place to start from though.

File: src/Guest/Guest.js [hunk 2]
```diff
@@ -2,12 +2,64 @@ import React, { Component } from "react";
 import "./Guest.scss";
 
 class Guest extends Component {
-  constructor() {
-    super()
+  constructor(props) {
+    super(props)
     this.state = {
-      
+// do we need it to be a class?
```
Comment (jordy1611) [inline src/Guest/Guest.js:8]: Is there ever going to be a state change with guest? If it was a generic User component I would say definitely say yes. Does guest need to even be a component? Or should it just be a property passed to different Components?

File: src/LandingPage/LandingPage.test.js [hunk 3]
```diff
@@ -12,10 +12,11 @@ describe('LandingPage', () => {
  })
 
   it('should render a greeting text and array of images', () => {
+    // how do we test for the larger welcoming text on landing page?
```
Comment (jordy1611) [inline src/LandingPage/LandingPage.test.js:15]: If it's what renders by default we should be able to get it with getByText without doing anything fancy? Maybe we have to do a mocked fetch that returns an empty set of data to simulate it?

**Review by jordy1611 — COMMENTED** [review]
Looks good. Really appreciate you taking the lead to set this all up. Think we're in a good spot to start adding features.

File: src/Guest/Guest.js (hunk 2 above)
Comment (JoshSevy) [reply src/Guest/Guest.js:8]: It depends on the direction we want to go with this project if we decide to use hooks we wouldn't need it to be a class component.

**Comment by JoshSevy** [comment]
This is a great start to this project, I think we could easily divide up functionality and have this done in no time. Great work! So many projects. When did we become legit developers? Haha

**Pull request merged**